crypto/s2n_ecc_evp.c

```diff
@@ -26,11 +26,11 @@
 DEFINE_POINTER_CLEANUP_FUNC(EVP_PKEY *, EVP_PKEY_free);
 DEFINE_POINTER_CLEANUP_FUNC(EVP_PKEY_CTX *, EVP_PKEY_CTX_free);
 
-/* IANA values can be found here:
- * https://tools.ietf.org/html/rfc8446#appendix-B.3.1.4 */
-/* Share sizes are described here:
- * https://tools.ietf.org/html/rfc8446#section-4.2.8.2 and include the extra
- * "legacy_form" byte */
+#if !S2N_IS_X25519_SUPPORTED
```

Comment:
Here and throughout: `S2N_IS_X25519_SUPPORTED` isn't really being used to indicate if x25519 is supported anymore. Instead it's being used as a proxy for "Am I on a modern version of libcrypto?" Modern libcrypto gives you not only x25519 but also many other useful/needed methods. Gating those methods (unrelated to x25519) on a macro called `S2N_IS_X25519_SUPPORTED` is really confusing.

Maybe this macro should be renamed to something like `MODERN_EC_SUPPORTED`?  (This review will be completed with me mentally substituting in `MODERN_EC_SUPPORTED`.)

Reply:
Your right, the macro was to capture whether we were on a modern version of libcrypto or not. Will rename to `MODERN_EC_SUPPORTED`.